src/app/util/af-event.cpp

```diff
@@ -0,0 +1,291 @@
+    do                                                                                                                             \
+    {                                                                                                                              \
+        (control).status = EMBER_EVENT_INACTIVE;                                                                                   \
+    } while (0)
+
+/** @brief Sets this ::EmberEventControl as inactive (no pending event).
+ */
+#define emberEventControlSetActive(control)                                                                                        \
+    do                                                                                                                             \
+    {                                                                                                                              \
+        (control).status = EMBER_EVENT_ZERO_DELAY;                                                                                 \
+    } while (0)
+
+#include "gen/af-gen-event.h"
+
+struct EmberEventData
+{
+    /** The control structure for the event. */
+    EmberEventControl * control;
+    /** The procedure to call when the event fires. */
+    void (*handler)(void);
+};
+
+// *****************************************************************************
+// Globals
+
+#ifdef EMBER_AF_GENERATED_EVENT_CODE
+EMBER_AF_GENERATED_EVENT_CODE
+#endif // EMBER_AF_GENERATED_EVENT_CODE
+
+#if defined(EMBER_AF_GENERATED_EVENT_CONTEXT)
+uint16_t emAfAppEventContextLength        = EMBER_AF_EVENT_CONTEXT_LENGTH;
+EmberAfEventContext emAfAppEventContext[] = { EMBER_AF_GENERATED_EVENT_CONTEXT };
+#endif // EMBER_AF_GENERATED_EVENT_CONTEXT
+
+const char * emAfEventStrings[] = {
+
+#ifdef EMBER_AF_GENERATED_EVENTS
+    EMBER_AF_GENERATED_EVENT_STRINGS
+#endif
+
+};
+
+EmberEventData emAfEvents[] = {
+
+#ifdef EMBER_AF_GENERATED_EVENTS
+    EMBER_AF_GENERATED_EVENTS
+#endif
+
+    { NULL, NULL }
+};
+
+const char emAfStackEventString[] = "Stack";
+
+// *****************************************************************************
+// Functions
+
+const char * emberAfGetEventString(uint8_t index)
+{
+    return (index == 0XFF ? emAfStackEventString : emAfEventStrings[index]);
+}
+
+static EmberAfEventContext * findEventContext(uint8_t endpoint, EmberAfClusterId clusterId, bool isClient)
+{
+#if defined(EMBER_AF_GENERATED_EVENT_CONTEXT)
+    uint8_t i;
+    for (i = 0; i < emAfAppEventContextLength; i++)
+    {
+        EmberAfEventContext * context = &(emAfAppEventContext[i]);
+        if (context->endpoint == endpoint && context->clusterId == clusterId && context->isClient == isClient)
+        {
+            return context;
+        }
+    }
+#endif // EMBER_AF_GENERATED_EVENT_CONTEXT
+    return NULL;
+}
+
+EmberStatus emberAfEventControlSetDelayMS(EmberEventControl * control, uint32_t delayMs)
```

Comment:
There are callsites of this and some of the other `emberAfEventControlSetDelay*` functions here in `util.c` that got commented out and can presumably get uncommented now.

Reply:
I thought those were stubbed.  No?

Reply:
I mean there are functions that are commented out in their entirety because they call this function and this function did not exist.

That said those functions are not called by anything, so maybe we can just remove them....  Followup is fine either way, to be clear!

Reply:
I think we can remove them

Reply:
chose removal